addon/components/o-s-s/text-area.ts

```diff
@@ -0,0 +1,46 @@
+import { assert } from '@ember/debug';
+import { action } from '@ember/object';
+import Component from '@glimmer/component';
+
+interface OSSTextAreaArgs {
+  rows?: number;
+  resize?: 'vertical' | 'horizontal' | 'both' | 'none';
+  value?: string;
+  disabled?: boolean;
+  errorMessage?: string;
+  placeholder?: string;
+  onChange?(value: string): void;
+}
+
+export default class OSSTextArea extends Component<OSSTextAreaArgs> {
+  constructor(owner: unknown, args: OSSTextAreaArgs) {
+    super(owner, args);
+
+    if (this.args.resize) {
+      assert(
+        '[component][OSS::TextArea] The @resize parameter should be a value of resize',
+        ['vertical', 'horizontal', 'both', 'none'].includes(this.args.resize)
+      );
+    }
+  }
+
+  get rows(): number {
+    return this.args.rows || 1;
+  }
+
+  get computedClass(): string {
+    const classes: string[] = [];
+    if (this.args.resize === 'vertical') classes.push('oss-textarea--resize-v');
+    else if (this.args.resize === 'horizontal') classes.push('oss-textarea--resize-h');
+    else if (this.args.resize === 'none') classes.push('oss-textarea--resize-none');
+
+    return classes.join(' ');
+  }
+
+  @action
+  _onChange(value: string): void {
+    if (this.args.onChange) {
+      this.args.onChange(value);
+    }
```

Comment:
```suggestion
  this.args.onChange?.(value);
```
You can simplify that :smile: